specs/local-context-spec.md

```diff
@@ -0,0 +1,274 @@
+
+Anticipating user error with `GLOBAL_CONTEXT_PATH`, AI systems SHOULD attempt to identify if the `GLOBAL_CONTEXT_PATH` includes the client context path and gracefully handling using that as the full directory.
+
+Because the global context path can include configuration, including context communication protocols, AI systems MUST use this configuration for global context configuration.
+
+### Static directory context
+
+Static directory context is sourcing context files and context configuration from a pre-determined location and not changing the context during the active work of the AI system.
+
+For static directory context support, AI systems MUST source context files and configuration from the root of the mounted directory or AI working directory if mounted directory is not supported by the AI system.
+
+Example, if the mounted directory is `~/myproject` then the AI system must look for context and configuration at `~/myproject/<clientContextPath>`.
+
+### Dynamic subdirectory context
+
+Dynamic subdirectory context is actively adjusting available context to the AI system based on the active work that the AI system is doing and where it is doing it. This method allows context to be colocated to where work is being done.
+
+As AI systems determine work to do, they select which files/folders to perform actions (e.g. edit a file or create a file in a directory). When the AI system selects where work needs to happen, the rightmost directory on the path becomes the AI working directory. The directories between this AI working directory and the mounted directory are the ancestor directories. Each of those directories must be sourced for context files and configuration.
+
+For example, if the mounted directory is `~/myproject` and the AI working directory is `~/myproject/src/components` then the AI system must look for context files and configuration at `~/myproject/src/components/.context`, `~/myproject/src/.context`, and `~/myproject/.context`.
+
+AI systems MAY cache these lookups to improve performance for many directory lookups.
+
+
+### Context file formats
+
+There are no active standards around context file formats. The following is the industry best current practices for context files formats while improved standards are being formalized.
+
+- **Plaintext files**
+  - `.txt` files represent plaintext files. AI systems MUST accept the raw context as context.
+
+- **Markdown files**
+  - `.md` and `.mdc` files represent markdown files. AI systems MUST accept both extensions and the raw context as context. If the file contains [front matter](https://jekyllrb.com/docs/front-matter/), AI systems MUST parse the front matter and use it as properties.
```

Comment:
Are we supporting any front matter language (yaml/toml/json), or only yaml? Although our examples use yaml, it's unclear if clients should support toml/json as well.